portal-ui/src/screens/Console/ServiceAccounts/ServiceAccounts.tsx

```diff
@@ -0,0 +1,348 @@
+
+import React from "react";
+import { createStyles, Theme, withStyles } from "@material-ui/core/styles";
+import Table from "@material-ui/core/Table";
+import TableBody from "@material-ui/core/TableBody";
+import TableCell from "@material-ui/core/TableCell";
+import TableHead from "@material-ui/core/TableHead";
+import TableRow from "@material-ui/core/TableRow";
+import Paper from "@material-ui/core/Paper";
+import Grid from "@material-ui/core/Grid";
+import api from "../../../common/api";
+import {
+  Button,
+  Drawer,
+  IconButton,
+  LinearProgress,
+  TableFooter,
+  TablePagination,
+  Toolbar
+} from "@material-ui/core";
+import Typography from "@material-ui/core/Typography";
+import DeleteIcon from "@material-ui/icons/Delete";
+import {
+  NewServiceAccount,
+  ServiceAccount,
+  ServiceAccountsList
+} from "./types";
+import { MinTablePaginationActions } from "../../../common/MinTablePaginationActions";
+import EditIcon from "@material-ui/icons/Edit";
+import AddServiceAccount from "./AddServiceAccount";
+import DeleteServiceAccount from "./DeleteServiceAccount";
+import CredentialsPrompt from "./CredentialsPrompt";
+
+const styles = (theme: Theme) =>
+  createStyles({
+    seeMore: {
+      marginTop: theme.spacing(3)
+    },
+    paper: {
+      // padding: theme.spacing(2),
+      display: "flex",
+      overflow: "auto",
+      flexDirection: "column"
+    },
+    addSideBar: {
+      width: "480px",
+      minWidth: "320px",
+      padding: "20px"
+    },
+    errorBlock: {
+      color: "red"
+    },
+    tableToolbar: {
+      paddingLeft: theme.spacing(2),
+      paddingRight: theme.spacing(0)
+    },
+    wrapCell: {
+      maxWidth: "200px",
+      whiteSpace: "normal",
+      wordWrap: "break-word"
+    }
+  });
+
+interface IServiceAccountsProps {
+  classes: any;
+}
+
+interface IServiceAccountsState {
+  records: ServiceAccount[];
+  totalRecords: number;
+  loading: boolean;
+  error: string;
+  deleteError: string;
+  addScreenOpen: boolean;
+  page: number;
+  rowsPerPage: number;
+  deleteOpen: boolean;
+  selectedServiceAccount: ServiceAccount | null;
+  showNewCredentials: boolean;
+  newServiceAccount: NewServiceAccount | null;
+}
+
+class ServiceAccounts extends React.Component<
+  IServiceAccountsProps,
+  IServiceAccountsState
+> {
+  state: IServiceAccountsState = {
+    records: [],
+    totalRecords: 0,
+    loading: false,
+    error: "",
+    deleteError: "",
+    addScreenOpen: false,
+    page: 0,
+    rowsPerPage: 10,
+    deleteOpen: false,
+    selectedServiceAccount: null,
+    showNewCredentials: false,
+    newServiceAccount: null
+  };
+
+  fetchRecords() {
+    this.setState({ loading: true }, () => {
+      const { page, rowsPerPage } = this.state;
+      const offset = page * rowsPerPage;
+      api
+        .invoke(
+          "GET",
+          `/api/v1/service_accounts?offset=${offset}&limit=${rowsPerPage}`
+        )
+        .then((res: ServiceAccountsList) => {
+          this.setState({
```

Comment:
just to consider that setState is async function, dont know if we want to wait til is donde to run the following code

```
          this.setState({ page: newPage }, () => {
              this.fetchRecords();
            });
```

in that case ^ should be added as a callback in this.setState(state, callback function)

Reply:
In this particular case it's already ok to set the state immediately, in other places in the call, callbacks are passed to set state